FWCore/ParameterSet/interface/PluginDescription.h

```diff
@@ -150,32 +152,65 @@ namespace edm {
       using CreatedType = PluginDescriptionAdaptorBase<typename T::CreatedType>;
       using Factory = edmplugin::PluginFactory<CreatedType*()>;
 
-      {
-        std::stringstream ss;
-        ss << dfh.section() << "." << dfh.counter();
-        std::string newSection = ss.str();
+      std::string const& pluginCategory = Factory::get()->category();
+
+      printSpaces(os, indentation);
+      os << "There are multiple possible different descriptions for this ParameterSet\n";
+      printSpaces(os, indentation);
+      os << "because it will be used by a helper plugin object contained inside the top level\n";
+      printSpaces(os, indentation);
+      os << "module plugin object and the type of the helper plugin object is configurable.\n";
+      printSpaces(os, indentation);
+      os << "Or if it is in a vector of ParameterSets it might be used by multiple\n";
+      printSpaces(os, indentation);
+      os << "helper plugin objects and each could be configured with a different plugin type.\n";
+      printSpaces(os, indentation);
+      os << "Each plugin type could allow a different set of configuration parameters.\n";
+      printSpaces(os, indentation);
+      os << "Each subsection of this section has one of the possible descriptions.\n";
+      printSpaces(os, indentation);
+      os << "All of these plugins are from the category \"" << pluginCategory << "\".\n";
+      printSpaces(os, indentation);
+      os << "The plugin type is specified by the parameter named \"" << typeLabel_ << "\".\n";
 
+      if (!dfh.brief()) {
+        os << "\n";
+      }
+
+      if (dfh.wasCategoryAlreadyPrinted(pluginCategory)) {
         printSpaces(os, indentation);
-        os << "Section " << newSection << " " << Factory::get()->category() << " Plugins description:\n";
-        if (!dfh.brief())
+        os << "*** The descriptions for this plugin category already started printing above! ***\n";
```

Comment:
If you kept track of the category numbering, you could print the number here.

Reply:
I made this change and just force pushed it. "(See Section xx.xx.xx)" now appears and references the section where the descriptions started printing earlier.